lib/kubernetes-deploy/kubernetes_resource/service.rb

```diff
@@ -3,11 +3,15 @@ module KubernetesDeploy
   class Service < KubernetesResource
     TIMEOUT = 7.minutes
 
-    SYNC_DEPENDENCIES = %w(Pod Deployment)
-    def sync(mediator)
+    def sync(cache)
       super
-      @related_deployments = selector.present? ? mediator.get_all(Deployment.kind, selector) : []
-      @related_pods = selector.present? ? mediator.get_all(Pod.kind, selector) : []
+      if exists? && selector.present?
```

Comment:
This is an unrelated optimization: don't bother grabbing the deployments and pods if we're not going to use them.